app/services/scrape.rb

```diff
@@ -1,117 +1,237 @@
 
+  require'resolv'
+  # require 'selenium-webdriver'
+
   module Tags
     SCRIPT = 'script'
-    LINK = 'links'
+    LINK = 'link'
+  end
+
+  module SubSource
     SRC = 'src'
     HREF = 'href'
   end
 
   module DataTypes
     PLUGINS = 'plugins'
+    MUPLUGINS = 'mu-plugins'
     THEMES = 'themes'
     JS = 'js'
     CLOUDFLARE = 'cloudflare'
   end
 
-
-  def self.filter_wp_urls(urls, logger)
+  def self.filter_wp_urls(urls, logger, test_id)
     url_html_version_map = Hash.new{ |h,k| h[k] = Hash.new }
     threads = []
-    logger.info "fetching file"
-    file_data = File.readlines(File.join(__dir__,'proxy.txt'), chomp: true)
-    puts file_data.count
-    for i in 0..5 do
-      puts file_data[i]
-      urls.each do |url|
-        puts url
-        threads << Thread.new(){
-          thread_block(url,url_html_version_map,file_data[i])
-        }
-      end
+    # _proxy = ProxyDatum.order('RANDOM()').first
+    urls.each do |url_id|
+      threads << Thread.new(){
+        thread_block(url_id, url_html_version_map, logger, test_id)
+      }
     end
     threads.each do |thread|
       thread.join
     end
     return url_html_version_map
   end
 
-  def self.thread_block(url, url_html_version_map,proxy_ip)
+  def self.thread_block(url_id, url_html_version_map, logger, test_id)
     begin
-      puts url + "    " + proxy_ip
-      RestClient.proxy = "http://" + proxy_ip
-      html = Nokogiri::HTML.parse(RestClient.get url)
-      _url = Url.where(url: url).first
-      if _url 
-        _url_id = _url.id
-        url_html_version_map[_url_id] = {:html => html, :version => _url.site_data_info.cms_version}
-      else
-        _version = check_wordpress_in_meta(html)
-        if _version
-          _url_id =  Url.import_urls([url])[0]
-          url_html_version_map[_url_id] = {:html => html, :version => _version}
-        end
+      url = Url.find(url_id)
+      html = Nokogiri::HTML.parse(RestClient.get (url.url + "?x=#{rand(999999)}"))
+      #File.write("/tmp/#{url.url}", html)
+      cms_and_version_hash = cms_and_version(html)  # fetching both cms type and its version together 
+      if cms_and_version_hash.present?
+        url.cms || url.update(:cms => cms_and_version_hash[:cms])
+        url_html_version_map[url_id] = {:html => html, :cms_version => cms_and_version_hash[:cms_version]}
       end
     rescue => e
-      puts "#{url}...#{e}"
+      logger.info "Test Id : #{test_id} Url: #{url.url} Error: #{e}"
     end
   end
 
+  def self.cms_and_version(html)
+    is_wordpress(html) || is_drupal(html) || is_shopify(html) || is_joomla(html)
+  end
+  
+  def self.is_wordpress(html)
+    wordpress_and_version_hash = check_wordpress_in_meta(html) || Hash.new
+    if !wordpress_and_version_hash.present? and check_wordpress_in_html(html)
+      wordpress_and_version_hash[:cms] = "wordpress"
+      version_from_resource = find_wordpress_version(html)
+      wordpress_and_version_hash[:cms_version] = version_from_resource
```

Comment:
for now, if you can't find the version in meta, don't populate it

The other checks aren't reliable enough